### ISSUE TITLE
No Update File Causes App to Hang

### ISSUE THREAD

**Issue by Haldrie** [saramibreak/UmdImageCreator#2]
If you try to read a disc that does not have a firmware update on it the program will fail to find the update file and then hang and refuse to do anything other then pop off various errors. I originally only saw this when trying to dump UMD-Video discs but then went to dump my copy of Ape Escape: On The Loose which does not have an update and got the same behavior.

**Comment by saramibreak** [comment]
> the program will fail to find the update file

What is "update file"? This app doesn't find it.

**Comment by Haldrie** [comment]
"Cannot open disc0:/PSP_GAME/SYSDIR/UPDATE/PARAM.SFO: result=0x80010002" If I press circle or x after that I get the same disc info and then the error "Cannot create ms0:/iso/UCUS-98609_disc.txt" and then the app just sits there until I either try pressing a button again at which point I get the same text on the screen again or press triangle to exit after it stops hanging up to let me exit. Another behavior I noticed on Little Big Planet and Ratchet & Clank: Size Matters it will dump the disc info but won't dump the disc itself simply acting as if I pressed X and saying "Done" after it dumps the info even though I press circle.

**Comment by saramibreak** [comment]
- fixed: GetParamSfo always return TRUE even if there is not /PSP_GAME/SYSDIR/UPDATE/PARAM.SFO
[UmdImageCreator.zip](https://github.com/saramibreak/UmdImageCreator/files/5799567/UmdImageCreator.zip)

**Comment by Haldrie** [comment]
The app won't start now. I'm running PRO 661 and it just gives me the error "The game could not be started. (800200D9)".

**Comment by saramibreak** [comment]
sorry.
[UmdImageCreator.zip](https://github.com/saramibreak/UmdImageCreator/files/5802591/UmdImageCreator.zip)

**Comment by Haldrie** [comment]
Well it dumped the info but just showed Done without dumping the disc as if I pressed X when I pressed Circle.

**Comment by saramibreak** [comment]
I started this app on PSP-1000 and 6.61 LME-2.3. It's no problem.

> Well it dumped the info

Were 4 text files (_disc.txt, _mainError.txt, _mainInfo.txt, _volDesc.txt) created?
Can you upload these files?

> but just showed Done without dumping the disc as if I pressed X when I pressed Circle.

No error message?

**Comment by Haldrie** [comment]
Yes the text files were created and are populated with the correct info I would expect and no there was no error message.
[EDIT]
I just switched to 661 LME and tried Ape Escape again and now I am getting an error.
"Creating UCUS-98609,iso (LBA)      640/365840[F:ExecReadDisc][L:44] GetLastError: 0,"
This disc seems to read using any other dumping method I have access to including other dumping tools (though ones that are faulty and leave out the last sector in many cases) so not sure why a read error is happening.
[EDIT2]
Nevermind I guess it was a legit read error it seems to be working now. I wonder why it has issues on PRO.
